tiledb/sm/filesystem/azure.h

```diff
@@ -283,33 +283,50 @@ class Azure {
 
   class AzureRetryPolicy final : public azure::storage_lite::retry_policy_base {
    public:
-    AzureRetryPolicy(const bool under_test)
-        : azure::storage_lite::retry_policy_base()
-        , under_test_(under_test) {
-    }
-
+    /**
+     * The SDK invokes this routine before each request to Azure. This returns
+     * a pair: a bool to indicate if we should make a request and a time
+     * interval to wait before starting the request.
+     */
     azure::storage_lite::retry_info evaluate(
         const azure::storage_lite::retry_context& context) const override {
-      const int32_t max_retries = under_test_ ? 1 : 3;
```

Comment:
I can't remember why I did this, but it was masking this bug in the unit tests. Removed.